src/engine.py

```
@@ -7870,7 +7976,7 @@ def _resolve_barge_in_min_ms(
         """Resolve effective barge-in min duration with local/pipeline-only first-turn fast path."""
         base_min = int(getattr(cfg, "pipeline_min_ms", 0) or getattr(cfg, "min_ms", 250)) if pipeline_mode else int(getattr(cfg, "min_ms", 250))
 
-        scope_applies = pipeline_mode or (provider_name == "local")
+        scope_applies = pipeline_mode or (self._get_provider_kind(provider_name) == "local")
```

Ruff (0.15.13)

[error] Undefined name `self`

(F821)